File: tests/fixtures/go/repos.go
package main

import "fmt"

type User struct {
	ID int
}

type UserRepository struct{}

func (r *UserRepository) FindUser(id int) User {
	return User{ID: id}
}

func (r *UserRepository) DeleteUser(id int) {
	fmt.Println(id)
}

type AuditLog struct{}

func (a AuditLog) DeleteUser(id int) {
	fmt.Println(id)
}

type Notifier interface {
	Send(text string)
}

type EmailNotifier struct{}

func (e *EmailNotifier) Send(text string) {
	fmt.Println(text)
}

type SmsNotifier struct{}

func (s SmsNotifier) Send(text string) {
	fmt.Println(text)
}

File: tests/fixtures/typescript/repos.ts
export interface User {
  id: number;
}

export class UserRepository {
  findUser(id: number): User {
    return { id };
  }

  async deleteUser(id: number): Promise<void> {
    console.log(id);
  }
}

export class AuditLog {
  deleteUser(id: number): void {
    console.log(id);
  }
}

export interface Notifier {
  send(text: string): void;
}

export class EmailNotifier implements Notifier {
  send(text: string): void {
    console.log(text);
  }
}

export class SmsNotifier implements Notifier {
  send(text: string): void {
    console.log(text);
  }
}
